src/main/java/opendcs/dao/PropertiesSqlDao.java

```diff
@@ -36,36 +37,38 @@
  * For backward compatibility we do not name the columns (some DECODES tables had
  * columns named 'name' and 'value').
  * The name of the id column varies. For this reason it must be passed to the read/write methods.
- * 
+ *
  * @author mmaloney Mike Maloney, Cove Software, LLC
  */
-public class PropertiesSqlDao 
-	extends DaoBase 
+public class PropertiesSqlDao
+	extends DaoBase
 	implements PropertiesDAI
 {
 	public PropertiesSqlDao(DatabaseConnectionOwner tsdb)
 	{
 		super(tsdb, "PropertiesSqlDao");
 	}
-	
+
 	@Override
-	public void readProperties(String tableName, String idColumn, 
+	public void readProperties(String tableName, String idColumn,
 		DbKey parentKey, Properties props)
 			throws DbIoException
 	{
-		String q = "select * from " + tableName + " where " + idColumn + " = " + parentKey;
-		
-		ResultSet rs = doQuery(q);
+		Objects.requireNonNull(props, "A valid properties object must be passed into this function.");
+		Objects.requireNonNull(tableName, "A valid table name is required.");
+		Objects.requireNonNull(parentKey, "A valid Database Identifier is required.");
+		Objects.requireNonNull(idColumn, "The id column must be valid.");
+		String q = "select * from " + tableName + " where " + idColumn + " = ?";
```

Comment:
Any chance the tableName or idColumn could be evil input?  Code is trusting jdbc PreparedStatement take care of that; proabably OK in this context.  Consider simple check for (letters and underscore only names).  i.e. validTableName(), validColumnName() could augument requireNonNull .   Could be applied to other code in this file PropertiesSqlDao.

Reply:
Yes, technically, but one would have to get the evil change through the pull request process.

This is a common, if not great, idom in jdbc when conditions require varying names or columns.